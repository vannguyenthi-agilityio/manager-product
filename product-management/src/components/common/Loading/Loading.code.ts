export const asDefault = `
import Loading from '@chakra-ui/react';

/**
 * Loading provide a visual cue that an action is processing awaiting a course of change or a result for page.
 *
 */

 <Loading />
`;
