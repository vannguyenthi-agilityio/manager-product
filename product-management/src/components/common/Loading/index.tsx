import { Flex, Spinner } from '@chakra-ui/react';

export interface ILoading {
  isCenter?: boolean;
  isFull?: boolean;
  size?: string;
}

const Loading = ({ size = 'md', isCenter = true, isFull = false }: ILoading) => {
  return (
    <Flex
      position={isCenter ? 'fixed' : 'relative'}
      top={0}
      bottom={0}
      left={0}
      right={0}
      h={isFull ? '100vh' : 'auto'}
      w={isFull ? '100vw' : 'auto'}
      alignItems='center'
      justifyContent='center'
    >
      <Spinner size={size} />
    </Flex>
  );
};

export default Loading;
