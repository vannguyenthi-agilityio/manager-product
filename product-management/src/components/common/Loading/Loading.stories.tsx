import Storybook from '@storybook/react';

// Components
import Loading from '.';

// Code
import { asDefault } from './Loading.code';

export default {
  title: 'Components/Loading',
  component: Loading,
  parameters: {
    componentSource: {
      code: asDefault
    }
  }
} as Storybook.Meta<typeof Loading>;

const Template: Storybook.StoryFn<typeof Loading> = (args) => {
  return (
    <Loading
      isFull={true}
      {...args}
    />
  );
};

export const LoadingDefault = Template.bind({});

LoadingDefault.args = {
  size: 'xl'
};

LoadingDefault.parameters = {
  componentSource: {
    code: asDefault
  }
};
