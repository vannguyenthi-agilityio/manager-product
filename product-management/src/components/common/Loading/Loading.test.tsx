// Libraries
import { render } from '@testing-library/react';

// Components
import Loading from '.';

describe('Loading component', () => {
  it('should render Loading component', () => {
    const { container } = render(<Loading />);

    const spinnerLoadPage = container.querySelector('.chakra-spinner');
    expect(spinnerLoadPage).toBeTruthy();
  });

  describe('jest Snapshot testing', () => {
    it('matches Loading Snapshot', () => {
      const { asFragment } = render(<Loading />);

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
