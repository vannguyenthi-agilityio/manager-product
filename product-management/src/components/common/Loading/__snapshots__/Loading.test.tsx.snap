// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component jest Snapshot testing matches Loading Snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-817ns3"
  >
    <div
      class="chakra-spinner css-1y7joxr"
    >
      <span
        class="css-8b45rq"
      >
        Loading...
      </span>
    </div>
  </div>
</DocumentFragment>
`;
